api/v1/verticadb_webhook.go

```diff
@@ -320,6 +322,43 @@ func (v *VerticaDB) validateEndpoint(allErrs field.ErrorList) field.ErrorList {
 	return allErrs
 }
 
+func (v *VerticaDB) hasValidSvcName(allErrs field.ErrorList) field.ErrorList {
+	vdbName := v.ObjectMeta.Name
+	// vdb name will be used as is for headless svc name,
+	// thus match regex for service name (DNS-1035 label)
+	r := regexp.MustCompile(`^[a-z]([\-a-z0-9]{0,61}[a-z0-9])?$`)
+	if !r.MatchString(vdbName) {
+		err := field.Invalid(field.NewPath("metadata").Child("name"),
+			v.ObjectMeta.Name,
+			fmt.Sprintf("vdb name is used as the headless service name, and hence must match regex %q",
+				`^[a-z]([\-a-z0-9]{0,61}[a-z0-9])?$`))
+		allErrs = append(allErrs, err)
+		// invalid vdb name does not necessarily imply invalid external service name
+		// but usually invalid vdb name results in invalid external service name for
+		// all subclusters, so prompt users to fix vdb name first and do not check
+		// external service names for now
+		return allErrs
+	}
+	// check external svc names
+	for i := range v.Spec.Subclusters {
+		sc := &v.Spec.Subclusters[i]
+		extSvcName := v.ObjectMeta.Name + "-" + sc.GetServiceName()
```

Comment:
There is already `hasValidDomainName` that checks that sc names are valid, so I think here you should just check `sc.ServiceName` and skip if it is empty.